### ISSUE TITLE
Bang methods should return the scope when successful, not just true

### ISSUE THREAD

**Issue by philomory** [makandra/consul#22]
Hey there,

I'm really liking Consul so far, however, there's an aspect of the auto-generated bang methods that I find rather frustrating.

I often-times want, in an index action,. to check if the user has access the resource being indexed; if they do, I want to populate the relevant variable with the list of resource instances they have access to but if they don't I want to raise Consul::Powerless so I can handle it in a rescue_from handler. Normally, I'd expect it to work like this:

``` ruby
@resources = current_power.resources!
```

...but what I actually need is more like:

``` ruby
@resources = current_power.resources if current_power.resources!
```

 I can't really use

``` ruby
@resources = current_power.resources
```

alone, either, because if they don't have access to that resource at all (that is, the method returns nil rather than simply an empty set), then I'll get a NoMethodError in the view when I try to iterate over `@resources`.

I can probably write up a pull request for this, but I thought I'd ask for your opinion on the matter first; there might be a logic to the current behavior that I don't understand, and I didn't want to code up a  pull request for something that turns out to be a bad idea.

**Comment by philomory** [comment]
As an addendum, this also applies to crud actions:
```ruby
@resource = current_power.updatable_resources!.find(params[:id])
```
would be better than
```ruby
@resource =  current_power.updateable_resources.find(params[:id]) if current_power.updatable_resources!
```

**Comment by henning-koch** [comment]
Although we have the same requirements, we don't run into that issue. Reason is that we use [Consul's idiomatic controller action protection](https://github.com/makandra/consul#protect-entry-into-controller-actions) like so:

```
class NotesController < ApplicationController

  power :notes, as: :note_scope

  def show
    note_scope.find(params[:id])
  end

end
```

Nonetheless, I will accept a PR that makes bang-methods returns scopes if it is well-written and has passing tests.

**Comment by philomory** [comment]
After thinking about it, I think I can restructure my code to use the idiomatic guard at least 90% of the time. That said, I still think returning the scope instead of true is a good idea, so I'll see if I can build a pull request for it.

**Comment by dastra-mak** [comment]
closed by pull #42

**Comment by dastra-mak** [comment]
@philomory this is now released as `0.13.2` (see https://github.com/makandra/consul/pull/43)
